clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -3156,12 +3156,21 @@ raw2trace_t::write(raw2trace_thread_data_t *tdata, const trace_entry_t *start,
             }
         }
     }
-    if (end > start &&
-        !tdata->out_file->write(reinterpret_cast<const char *>(start),
-                                reinterpret_cast<const char *>(end) -
-                                    reinterpret_cast<const char *>(start))) {
-        tdata->error = "Failed to write to output file";
-        return false;
+    if (end > start) {
+        // We reach here also when we're not writing to a zip archive.
```

Comment:
What do you mean "also" -- do we come into here for a zip output too?  If so won't the instr count then be wrong?

Reply:
I thought in some cases there may be some entries pending to be written to the archive. Otherwise I'd've expected this logic to be part of an `else` clause for `if (tdata->out_archive != nullptr)`, instead of a separate if condition. Is that not the case?

Reply:
Oh wait. It is overcounting. I did not re-verify the zip instr count after I made this part of the change in 62c085ead6bb80c8c0e4b7b8a341aafc2318406e. Let me fix this.

Reply:
I was overcounting the last chunk's data. Fixed it.

Reply:
But control indeed reaches here even for zip output -- for the last chunk.